File: spl-net/src/main/java/bgu/spl/net/impl/messages/NotificationMessage.java
package bgu.spl.net.impl.messages;

public class NotificationMessage extends Message{

    private char pm_public;
    private String postingUser;
    private String content;
    public NotificationMessage(char pm_public,String postingUser,String content) {
        super((short)9);
        this.pm_public=pm_public;
        this.postingUser=postingUser;
        this.content=content;
    }

    public char getPm_public(){return pm_public;}

    public String getPostingUser(){return postingUser;}

    public String getContent(){return content;}

    public String toString(){
        String s="09";
        s+=pm_public;
        s+=postingUser;
        s+='\0';
        s+=content;
        s+='\0';
        return s;}
}

File: Boost_Echo_Client/src/NotificationMessage.cpp
//
// Created by MAY on 04/01/2022.
//

#include "../include/NotificationMessage.h"

NotificationMessage::NotificationMessage(char _pm_public,  std::string _postingUser, std::string _content,  short _opCode) : Message(_opCode), pm_public(_pm_public), postingUser(_postingUser), content(_content) {
}

char NotificationMessage::getPm_public() const {
    return pm_public;
}

std::string NotificationMessage::getPostingUser() {
    return postingUser;
}

std::string NotificationMessage::getContent() {
    return content;
}

std::string NotificationMessage::toString() {
    std::string answer="NOTIFICATION ";
    answer.push_back(getPm_public());
    answer.push_back(' ');
    answer.append(getPostingUser());
    answer.push_back(' ');
    answer.append(getContent());
    return answer;
}
